### DISCUSSION TITLE
Multiple node selection by mouse

### DISCUSSION THREAD

**Discussion by ismyhairnice** [xyflow/xyflow#1827] [Q&A]
Hi all

I understand that react-flow support multiSelectionKeyCode by holding the key and click on the desire nodes to select. Just wondering if there is any chance selecting of nodes can be done by click on the mouse, hold and drag across the nodes to select?

**Comment by moklick** [comment]
Do you mean the multi selection by pressing shift?

**Reply by visualsitemaps** [reply]
@moklick i think he means this: https://share.getcloudapp.com/2NuvWgyk
any tips for this? thanks!

**Comment by doflo-dfa** [comment]
I think this is already possible ... press and hold shift and then left click wait a beat and start dragging ... for me it draws a pretty blue selection box ... at least on v10

**Reply by doflo-dfa** [reply]
I would like to tweak the amount of time I have to wait before starting my drag as I have seen that frustrate users in testing.

**Reply by bilogic** [reply]
i think this solves the problem better https://codesandbox.io/p/sandbox/figma-controls-vyvrvp?file=%2Fsrc%2FFlow.tsx%3A9%2C3-9%2C16
